.github/workflows/pull-request-lint.yaml

```diff
@@ -0,0 +1,39 @@
+name: Pull Request Linting
+on:
+  # Trigger the workflow when PRs pushed; we still run the workflow files from
+  # the master branch, to ensure that malicious actors cannot exfiltrate any
+  # secrets.
```

Comment:
One could still attack some vulnerability of the linters themselves, to run arbitrary code on PR push. But I suppose that's a small enough attack surface.

Reply:
Yes, but the linter runs don't get to have any secrets (since we don't need to create clusters to test them, and we check out the source with `persist-credentials: false`).  The issue with the workflow file is that _it_ can have access to secrets (e.g. `github.token`), which it may choose to export to scripts.

…I think; it'll probably be a good idea to double check the environment in that case.  I'll try to do that.  Either way it sounds like I need to expand the comment to say ↑.

Reply:
Added more comments in fcb37225b6f4f4e2c9dc674b038ddf60abbd3580